File: android/src/main/java/com/capacitor/plugin/inappbrowser/InAppBrowserPlugin.java
package com.capacitor.plugin.inappbrowser;

import android.net.Uri;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "InAppBrowser")
public class InAppBrowserPlugin extends Plugin {

    private InAppBrowser implementation;

    @Override
    public void load() {
        implementation = new InAppBrowser(getContext(), getActivity(), this::onInAppBrowserEvent);
    }

    @PluginMethod
    public void open(PluginCall call) {
        String url = call.getString("url");
        if (url == null || url.isEmpty()) {
            call.reject("URL must be non null nor empty.");
            return;
        }

        implementation.open(Uri.parse(url));
        call.resolve();
    }

    @PluginMethod
    public void close(PluginCall call) {
        String id = call.getString("id");
        if (id == null || id.isEmpty()) {
            call.reject("Id must be non null nor empty.");
            return;
        }

        implementation.close(id);
        call.resolve();
    }

    void onInAppBrowserEvent(int event) {
        switch (event) {
            case InAppBrowser.BROWSER_OPEN_EVENT:
                notifyListeners("open", null);
                break;
            case InAppBrowser.BROWSER_CLOSE_EVENT:
                notifyListeners("close", null);
                break;
        }
    }
}

File: ios/Plugin/InAppBrowserPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(InAppBrowserPlugin)
public class InAppBrowserPlugin: CAPPlugin {
    private let implementation = InAppBrowser()
    
    @objc func open(_ call: CAPPluginCall) {
        guard let urlString = call.getString("url"), let url = URL(string: urlString) else {
            call.reject("Must provide a valid URL to open")
            return
        }
        
        implementation.open(url)
        call.resolve()
    }
    
    @objc func close(_ call: CAPPluginCall) {
        guard let id = call.options["id"] as? String else {
          call.reject("Must provide an id")
          return
        }
        
        implementation.close(id)
        call.resolve()
    }
}
